### PR TITLE
Distorbutedstorage documentation

### PR DESCRIPTION
**Pull request by LiZhenCheng9527**
**What type of PR is this?**
Provide documentation on the use of distorbutedstorage

### CODE REVIEW

File: docs/content/en/docs/fleet-manager/distributedstorage-plugin.md [hunk 1]
```diff
@@ -0,0 +1,416 @@
+---
+title: "Enable Policy Management with fleet"
+linkTitle: "Policy Management"
+weight: 30
+description: >
+  The easiest way to enable policy management with fleet.
```
Comment (zirain) [inline docs/content/en/docs/fleet-manager/distributedstorage-plugin.md:6]: ?

File: docs/content/en/docs/fleet-manager/distributedstorage-plugin.md [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+
+In this tutorial, we will cover how to implement unified distributed storage on a set of clusters, using [Fleet](https://kurator.dev/docs/references/fleet-api/#fleet)
+
+## Architecture
+
+Fleet's unified distributed storage is built on top of [Rook](https://rook.io/), and the overall architecture is shown below:
+
+{{< image width="100%"
+    link="./image/fleet-policy.drawio.svg"
+    >}}
```
Comment (zirain) [inline docs/content/en/docs/fleet-manager/distributedstorage-plugin.md:17]: this's wrong.

File: docs/content/en/docs/fleet-manager/distributedstorage-plugin.md [hunk 3]
```diff
@@ -0,0 +1,416 @@
+---
+title: "Enable Policy Management with fleet"
+linkTitle: "Policy Management"
+weight: 30
```
Comment (Xieql) [inline docs/content/en/docs/fleet-manager/distributedstorage-plugin.md:4]: consider change `weight` from 30 to 35 or 50

File: docs/content/en/docs/fleet-manager/distributedstorage-plugin.md [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+apiVersion: fleet.kurator.dev/v1alpha1
+kind: Fleet
+metadata:
+  name: quickstart
+  namespace: default
+spec:
+  clusters:
+    - name: kurator-member1
+      kind: AttachedCluster
+    - name: kurator-member2
```
Comment (Xieql) [inline docs/content/en/docs/fleet-manager/distributedstorage-plugin.md:81]: miss `kind`, it is a required setting

File: docs/content/en/docs/fleet-manager/distributedstorage-plugin.md [hunk 5]
```diff
@@ -0,0 +1,416 @@
...
+    - name: data0
+      replicated:
+        size: 3
+  preserveFilesystemOnDelete: true
+  metadataServer:
+    activeCount: 1
+    activeStandby: true
+```
+
+The Rook operator will create all the pools and other resources necessary to start the service. To confirm the filesystem is configured, wait for the mds pods to start:
```
Comment (Xieql) [inline docs/content/en/docs/fleet-manager/distributedstorage-plugin.md:233]: NIT: maybe you can explain what is `mds` here. this is the first place where `mds` is used

File: docs/content/en/docs/fleet-manager/distributedstorage-plugin.md (hunk 2 above)
Comment (hzxuzhonghu) [reply docs/content/en/docs/fleet-manager/distributedstorage-plugin.md:17]: good catch


@LiZhenCheng9527 You can read the preview link

File: docs/content/en/docs/fleet-manager/distributedstorage-plugin.md [hunk 6]
```diff
@@ -0,0 +1,416 @@
...
+
+The easiest way to do this is to mount an Raw disk on the nodes.
+
+### 3. Secrets and Setup for Attached Clusters
+
+1. Running the following command to create two secrets to access attached clusters.
+
+```console
+kubectl create secret generic kurator-member1 --from-file=kurator-member1.config=/root/.kube/kurator-member1.config
+kubectl create secret generic kurator-member2 --from-file=kurator-member2.config=/root/.kube/kurator-member2.config
```
Comment (hzxuzhonghu) [inline docs/content/en/docs/fleet-manager/distributedstorage-plugin.md:43]: It is unclear what this is operating to others.

AFAIK, there is a prerequisite, a host cluster is already installed. SO you should add the clusters setup as a  prerequisite

File: docs/content/en/docs/fleet-manager/distributedstorage-plugin.md [hunk 7]
```diff
@@ -0,0 +1,416 @@
...
+    key: kurator-member2.config
+EOF
+```
+
+## Create a Fleet with the DistributedStorage Plugin Enabled
+
+Run following command to create rook operator and rook ceph cluster in the Fleet:
+
+```console
+apiVersion: fleet.kurator.dev/v1alpha1
```
Comment (hzxuzhonghu) [inline docs/content/en/docs/fleet-manager/distributedstorage-plugin.md:72]: missing kubectl apply ?

File: docs/content/en/docs/fleet-manager/distributedstorage-plugin.md [hunk 8]
```diff
@@ -0,0 +1,416 @@
...
+      storage:
+        dataDirHostPath: /var/lib/rook
+        monitor:
+          count: 3
+          labels:
+            role: MonitorNodeLabel
+        manager:
+          count: 2
+          annotations:
+            role: ManagerNodeAnnotation
```
Comment (hzxuzhonghu) [inline docs/content/en/docs/fleet-manager/distributedstorage-plugin.md:93]: there is no device setting, what real volume does it use?

File: docs/content/en/docs/fleet-manager/distributedstorage-plugin.md [hunk 9]
```diff
@@ -0,0 +1,416 @@
...
+
+```
+
+### Fleet DistributedStorage Plugin Configuration Explained
+
+Let's delve into the `spec` section of the above Fleet:
+
+- `clusters`: Contains the two `AttachedCluster` objects created earlier, indicating that the distributedstorage plugin will be installed on these two clusters.
+- `plugin`: The `distributedStorage` indicates the description of a distributedstorage plugin. For more configuration options, please refer to the [Fleet API](https://kurator.dev/docs/references/fleet-api/).
```
Comment (hzxuzhonghu) [inline docs/content/en/docs/fleet-manager/distributedstorage-plugin.md:102]:
```suggestion
- `plugin`: The `distributedStorage` indicates the configuration of a distributedstorage plugin. For more configuration options, please refer to the [Fleet API](https://kurator.dev/docs/references/fleet-api/).
```

File: docs/content/en/docs/fleet-manager/distributedstorage-plugin.md [hunk 10]
```diff
@@ -0,0 +1,416 @@
...
+rook-ceph-osd-0-996475cb6f-cfbwh                                    1/1     Running            0          32h
+rook-ceph-osd-1-67f5ff649c-vs5qt                                    1/1     Running            0          32h
+rook-ceph-osd-2-7d4c78b74b-l8jg5                                    1/1     Running            0          32h
+rook-ceph-osd-prepare-testpool-dev-linux-0001-jz766                 0/1     Completed          0          32h
+rook-ceph-osd-prepare-testpool-dev-linux-0003-t47st                 0/1     Completed          0          32h
+rook-ceph-osd-prepare-testpool-dev-linux-0004-kw94n                 0/1     Completed          0          32h
+rook-ceph-rgw-ceph-objectstore-a-5c4df48bbb-bf6jn                   2/2     Running            0          32h
+```
+
+## Storage Configuration
```
Comment (hzxuzhonghu) [inline docs/content/en/docs/fleet-manager/distributedstorage-plugin.md:147]: how about to be more clear
```suggestion
## Persistent Volume Use Guide
```

File: docs/content/en/docs/fleet-manager/distributedstorage-plugin.md [hunk 11]
```diff
@@ -0,0 +1,416 @@
...
+kind: StorageClass
+metadata:
+   name: rook-ceph-block
+provisioner: rook-ceph.rbd.csi.ceph.com
+parameters:
+    clusterID: rook-ceph
+    pool: replicapool
+    imageFormat: "2"
+    imageFeatures: layering
+
```
Comment (hzxuzhonghu) [inline docs/content/en/docs/fleet-manager/distributedstorage-plugin.md:187]: remove

File: docs/content/en/docs/fleet-manager/distributedstorage-plugin.md [hunk 12]
```diff
@@ -0,0 +1,416 @@
...
+    csi.storage.k8s.io/provisioner-secret-name: rook-csi-rbd-provisioner
+    csi.storage.k8s.io/provisioner-secret-namespace: rook-ceph
+    csi.storage.k8s.io/controller-expand-secret-name: rook-csi-rbd-provisioner
+    csi.storage.k8s.io/controller-expand-secret-namespace: rook-ceph
+    csi.storage.k8s.io/node-stage-secret-name: rook-csi-rbd-node
+    csi.storage.k8s.io/node-stage-secret-namespace: rook-ceph
+    csi.storage.k8s.io/fstype: ext4
+
```
Comment (hzxuzhonghu) [inline docs/content/en/docs/fleet-manager/distributedstorage-plugin.md:195]: remove

File: docs/content/en/docs/fleet-manager/distributedstorage-plugin.md [hunk 13]
```diff
@@ -0,0 +1,416 @@
...
+```console
+kubectl get po -A --kubeconfig=/root/.kube/kurator-member1.config
+kubectl get po -A --kubeconfig=/root/.kube/kurator-member2.config
+```
+
+Perhaps there are still some rook components left over, which can be executed by running the following command:
+
+```console
+kubectl api-resources --verbs=list --namespaced -o name | xargs -n 1 kubectl get --show-kind --ignore-not-found -n rook-ceph
```
Comment (hzxuzhonghu) [inline docs/content/en/docs/fleet-manager/distributedstorage-plugin.md:401]: This is running based on the attached cluster, so you should make it explicit

File: docs/content/en/docs/fleet-manager/distributedstorage-plugin.md [hunk 14]
```diff
@@ -0,0 +1,416 @@
...
+      storage: 1Gi
+EOF
+```
+
+You can use different types of storage by changing the name of the storage class corresponding to the `spec.storageclassname`.
+
+Reference the created Persistent Volume Claims to the Pod for use. Here is an example of how to use it:
+
+```console
+kubectl apply -f -<<EOF
```
Comment (hzxuzhonghu) [inline docs/content/en/docs/fleet-manager/distributedstorage-plugin.md:353]: let's consistently use kurator application to create the pod/pvc

File: docs/content/en/docs/fleet-manager/image/distributedstorage.svg
Comment (Xieql) [inline docs/content/en/docs/fleet-manager/image/distributedstorage.svg:1]: the image is not right. When exporting in SVG format, choose to `embed images` and `convert tags to SVG format`

File: docs/content/en/docs/fleet-manager/distributedstorage-plugin.md [hunk 15]
```diff
@@ -0,0 +1,379 @@
...
+EOF
+```
+
+### Fleet DistributedStorage Plugin Configuration Explained
+
+Let's delve into the `spec` section of the above Fleet:
+
+- `clusters`: Contains the two `AttachedCluster` objects created earlier, indicating that the distributedstorage plugin will be installed on these two clusters.
+- `plugin`: The `distributedStorage` indicates the configuration of a distributedstorage plugin. `dataDirHostPath` defines the directory in which the rook-ceph cluster will save the cluster configuration. `monitor` and `manager` provide configuration for the mon and mgr components of ceph. For more configuration options, please refer to the [Fleet API](https://kurator.dev/docs/references/fleet-api/).
+- In addition to the configuration fields mentioned above, kurator also provides `storage.devices` field that allows you to specify the use of disks mounted in a specific directory such as `sda` in the cluster. And `storage.nodes` field that allows you to specify the use of storage resources on specific nodes. For more configuration options, please refer to the [Fleet API](https://kurator.dev/docs/references/fleet-api/).
```
Comment (hzxuzhonghu) [inline docs/content/en/docs/fleet-manager/distributedstorage-plugin.md:81]: I am not sure what you mean by `sda` in the cluster? Is it set in the example?

File: docs/content/en/docs/fleet-manager/distributedstorage-plugin.md [hunk 16]
```diff
@@ -0,0 +1,366 @@
...
+    clusterID: rook-ceph
+    pool: replicapool
+reclaimPolicy: Delete
+allowVolumeExpansion: true
+```
+
+There are a few things to note in the above block storage class configuration:
+
+- provisioner is configured in the format (operator-namespace).rbd.csi.ceph.com. Change "rook-ceph" provisioner prefix to match the operator namespace if needed.
+- `parametes.clusterID` is the namespace where the rook cluster is running.
```
Comment (hzxuzhonghu) [inline docs/content/en/docs/fleet-manager/distributedstorage-plugin.md:171]: clusterID is namespace?

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm

please respond the comments below each one.  And this is missing examples for filesystem and object storage

File: docs/content/en/docs/fleet-manager/distributedstorage-plugin.md (hunk 8 above)
Comment (LiZhenCheng9527) [reply docs/content/en/docs/fleet-manager/distributedstorage-plugin.md]: Without specifying a device, rook's osd-prepare job scans the node for all devices that can be retrieved with command `lsflk --all`.

File: docs/content/en/docs/fleet-manager/distributedstorage-plugin.md (hunk 15 above)
Comment (LiZhenCheng9527) [reply docs/content/en/docs/fleet-manager/distributedstorage-plugin.md]: sda is a device mounted on `/dev/sda`. if a specific mount path is specified, rook will check the specified path on all nodes in the cluster for the presence of storage resources that comply with requirements.

File: docs/content/en/docs/fleet-manager/distributedstorage-plugin.md (hunk 16 above)
Comment (LiZhenCheng9527) [reply docs/content/en/docs/fleet-manager/distributedstorage-plugin.md:171]: In the documentation for rook, it is mentioned that the clusterID is the namespace where the rook cluster is running.
You can refer to the following link [rook documtation](https://rook.io/docs/rook/v1.11/Storage-Configuration/Block-Storage-RBD/block-storage/#provision-storage)

**Pull request merged**